app/models/job_proxy_dispatcher.rb

```diff
@@ -137,59 +67,8 @@ def do_dispatch(job, ems_id)
     )
   end
 
-  def queue_signal(job, options)
-    Benchmark.current_realtime[:queue_signal_count] += 1
-    Benchmark.realtime_block(:queue_signal) do
-      return if options.blank?
-      default_opts = {
-        :class_name  => "Job",
-        :method_name => "signal",
-        :instance_id => job.id,
-        :task_id     => job.guid,
-        :priority    => MiqQueue::HIGH_PRIORITY,
-        :role        => "smartstate"
-      }
-
-      default_opts[:zone] = job.zone if job.zone
-      options = default_opts.merge(options)
-      # special case signal(:abort) - so we can easily pull off the queue
-      if (sig = options[:args].first) == :abort
-        options[:args].shift # remove :abort from args
-        options[:method_name] = "signal_abort"
-      end
-      MiqQueue.put_unless_exists(options) do |msg|
-        _log.warn("Previous Job signal [#{sig}] for Job: [#{job.guid}] is still running, skipping...") unless msg.nil?
-      end
-    end
-  end
-
-  def start_job_on_proxy(job, proxy)
-    assign_proxy_to_job(proxy, job)
-    _log.info("Job #{job.attributes_log}")
-    job_options = {:args => ["start"], :zone => MiqServer.my_zone, :server_guid => proxy.guid, :role => "smartproxy"}
-    @active_vm_scans_by_zone[MiqServer.my_zone] += 1
-    queue_signal(job, job_options)
-  end
-
-  def assign_proxy_to_job(proxy, job)
-    job.miq_server_id   = proxy.id
-    job.started_on      = Time.now.utc
-    job.dispatch_status = "active"
-    job.save
-
-    # Increment the counts for busy proxies and busy hosts for embedded
-    busy_proxies["MiqServer_#{job.miq_server_id}"] ||= 0
-    busy_proxies["MiqServer_#{job.miq_server_id}"] += 1
-
-    # Track the host/vc resource for embedded scans so we can limit the resource impact
-    if (key = embedded_scan_resource(@vm))
-      busy_resources_for_embedded_scanning[key] ||= 0
-      busy_resources_for_embedded_scanning[key] += 1
-    end
-  end
-
   def self.waiting?
-    Job.where(:state => "waiting_to_start").or(Job.where(:type => "InfraConversionJob").where.not(:state => ["finished", "waiting_to_start"]))
```

Comment:
This would include _all_ job not just jobs that were handled by the job proxy dispatcher (hold-over from VmScan being the only kind of Job?)

Reply:
Yeah, that could be a problem.  It was originally written as the only entrypoint to start a job.  If it's no longer the case, we should certainly change this assumption.

Reply:
Yeah that hasn't been the case for a few years now, so this inadvertently fixes that :sweat_smile: